src/UseCase/Response/Response.php

```diff
@@ -1,150 +1,104 @@
 <?php
 
-declare(strict_types = 1);
+declare(strict_types=1);
 
 namespace Damianopetrungaro\CleanArchitecture\UseCase\Response;
 
-use Damianopetrungaro\CleanArchitecture\Common\Collection\CollectionInterface;
-use Damianopetrungaro\CleanArchitecture\UseCase\Error\ErrorInterface;
+use Damianopetrungaro\CleanArchitecture\UseCase\Error\Error;
 
-class Response implements ResponseInterface
+interface Response
 {
     /**
-     * Available Response status
-     */
-    protected const STATUS_FAILED = 'FAILED';
-    protected const STATUS_SUCCESSFUL = 'SUCCESSFUL';
-
-    /**
-     * @var string $status
-     */
-    private $status;
-
-    /**
-     * Contains the error list
-     * (This list contains of data to show in case of failure)
+     * Add data to the data list.
      *
-     * @var CollectionInterface $errors
-     */
-    private $errors;
-
-    /**
-     * Contains the data list
-     * (This list contains of data to show in case of success)
+     * @param $key
+     * @param $value
      *
-     * @var CollectionInterface $data
+     * @return void
      */
-    private $data;
+    public function addData($key, $value): void;
 
     /**
-     * Response constructor.
+     * Add an Error to the error list.
      *
-     * @param CollectionInterface $data
-     * @param CollectionInterface $errors
-     */
-    public function __construct(CollectionInterface $data, CollectionInterface $errors)
-    {
-        $this->data = $data;
-        $this->errors = $errors;
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function addData($key, $value) : void
-    {
-        $values = $this->data->get($key, []);
-        $values[] = $value;
-        $this->data = $this->data->with($values, $key);
-    }
-
-    /**
-     * {@inheritDoc}
+     * @param $key
+     * @param Error $error
+     *
+     * @return void
      */
-    public function addError($key, ErrorInterface $error) : void
-    {
-        $values = $this->errors->get($key, []);
-        $values[] = $error;
-        $this->errors = $this->errors->with($values, $key);
-    }
+    public function addError($key, Error $error): void;
 
     /**
-     * {@inheritDoc}
+     * Return the data list content.
+     *
+     * @return array
      */
-    public function getData() : array
-    {
-        return $this->data->all();
-    }
+    public function getData(): array;
```

Comment:
What king of data this returns? the request's body? the headers?

Reply:
Return the data (no-http here as mentioned in the reply) that your use case expects to return.